pallets/protos/src/lib.rs

```diff
@@ -535,6 +538,46 @@ pub mod pallet {
 			<ProtosByTag<T>>::get(&tags)
 		}
 
+		fn is_proto_having_any_tags(proto_hash: &Hash256, tags: &Vec<Tags>) -> bool {
+			if let Some(struct_proto) = <Protos<T>>::get(proto_hash) {
+				tags.into_iter().any(|tag| struct_proto.tags.contains(&tag))
+			} else {
+				false
+			}
+		}
+
+		pub fn get_by_tags(tags: Vec<Tags>, owner: Option<T::AccountId>, limit: u32) -> Option<Vec<Hash256>> {
+
+			match owner {
+				Some(owner) => {
+					if let Some(vector_protos) = <ProtosByOwner<T>>::get(ProtoOwner::User(owner)) {
+						let iter_protos = vector_protos.into_iter();
+						let iter_protos_filtered = iter_protos.filter(|proto| Self::is_proto_having_any_tags(proto, &tags));
+						let iter_protos_limited = iter_protos_filtered.take(limit as usize);
+						Some(iter_protos_limited.collect::<Vec<Hash256>>())
+					} else {
+						None
+					}
+				},
+				None => {
+					/// TODO - Consider improving the closure in map
+					let iter_protos = tags.into_iter().map(|tag| <ProtosByTag<T>>::get(&tag).unwrap_or(Vec::new())).flatten();
```

Comment:
Not efficient, you are getting all? Does this return an iterator?
I think you need to use StorageMap calls better, there are iterators maybe to investigate

Reply:
Okay if we want `StorageMap` to return an iterator, I will need to change it to `IteratorStorageMap`

Reply:
Yes!